### DISCUSSION TITLE
Tools/CLI

### DISCUSSION THREAD

**Discussion by DiceMasters** [dcrtit/kit#9] [Ideas]
Ряды наши редеют, поэтому вспоминая хронологию нашего разговора в среду (02.02.2022), предлагаю к рассмотрению и обсуждению инструменты с помощью которых можно запилить cli для генерации scss и компонентов.

Из того, что упоминалось:

- yeoman (https://yeoman.io)
- node

Вобще предлагаю для тех, кто уже знаком с yeoman (а это вроде только Леня), ознакомиться с этой статьей:
https://www.twilio.com/blog/how-to-build-a-cli-with-node-js
и в последствии оставить свой фидбэк, касательно плюсов и минусов.

Если у кого то будут свои темы для обсуждения генераторов, прошу сюда же.

**Comment by lennoximus** [comment]
Сегодня в течение дня посмотрю и дам фидбек.

**Comment by lennoximus** [comment]
Посмотрел. По моему мнению, полезнее и удобнее будет сделать примерно в том же ключе, как в статье, что ты скинул, - больше контроля + есть возможность пощупать экосистему ноды, так сказатьб, расширить кругозор.

Надо только обсудить сам функционал: что будем генерировать, какое API будет у этого CLI-генератора и другие подобные моменты.

**Comment by DiceMasters** [comment]
Предлагаю (точнее что сейчас пытаюсь реализовать):
create-vue-component (cvc)
Параметры: 
- --setup -S (параметр устанавливает следующие аргументы по дефолту)
- *name* (название компонента)
- --lang -l = 'ts' | 'js' (String) (JS или TS)
- --class-component -cc
- --style -s = *css | scss | sass | less | stylus* (файл стилей, по дефолту scss)
- --help -h
- --version -v = 'vue3' | 'vue2' (String)(по дефолту vue2)

Дефолтные параметры находятся в .env

Пример:
`cvc --lang 'ts' --cc -s 'stylus'`

Жду еще предложений по доработке

**Reply by lennoximus** [reply]
А какие еще тулзы мы можем представить в будущем? Будем создавать только vue-компоненты? Или еще что-то?

мб сделать что-то с более общим названием типа `cli`, а там уже будет что-то вроде `generate --type=component {все опции что ты описал}`. Что думаешь?

**Reply by DiceMasters** [reply]
cli что то слишком общее название, я думаю пока над ним особо не думать, чисто, чтоб меньше времени занимало оставлю пока cvc

Вобще я еще туда думаю запилить, что то типа 
- --root-tag (-rt) (String) - корневой тег

**Reply by lennoximus** [reply]
В шаблоне?

**Reply by DiceMasters** [reply]
> В шаблоне?

Так точно, и плюс 

- --prefix (-prf) (String) - префикс для класса стилей и name'а типа как 'c-' в '.c-block'

**Reply by lennoximus** [reply]
Добро.
